### PR TITLE
database directory configuration

### PR DESCRIPTION
**Pull request by larskuhtz**
* resolves a long standing bug that places the pact db in the wrong place when a custom database location is configured.
* finally removes the long deprecated NodeId type from the code base. 

# TODO

verify that

* [x] `--node-id` isn't included in help messages, and results in warning when used
* [x] `--print-config` doesn't not show `nodeId`
* [x] configuring `nodeId` in config files result in warning    
* [x] database configuration bug is fixed
* [x] Database migrations work as expected

# Testing

* [x] start test chainweb-node with empty default location
     * [x] restart on with default location
     * [x] restart with non-existing custom location without trailing '/'
     * [x] restart with (now) existing custom location without trailing '/'
     * [x] restart with non-existing custom location with trailing '/'
     * [x] restart with (now) existing custom location with trailing '/'

* [x] start chainweb-node-2.0 (old) with default location
    * [x] start test chainweb-node with default location
    * [x] restart test chainweb-node with default location
    * [x] start test chainweb-node with custom location without trailing '/'
    * [x] start test chainweb-node with custom location with trailing '/'

* [x] start chainweb-node-2.0 (old) with custom location without trailing '/'
    * [x] start test chainweb-node with default location
    * [x] restart test chainweb-node with default location
    * [x] restart test chainweb-node with custom location without trailing '/'

* [x] start chainweb-node-2.0 (old) with custom location with trailing '/'
    * [x] start test chainweb-node with custom location with trailing '/'
    * [x] restart test chainweb-node with custom location with trailing '/'
    * [x] restart test chainweb-node with default location

* [x] start chainweb-node-2.0 (old) with custom location without trailing '/'
    * [x] start test chainweb-node with custom location with trailing '/'
    * [x] restart test chainweb-node with custom location with trailing '/'

* [x] start chainweb-node-2.0 (old) with custom location with trailing '/'
    * [x] start test chainweb-node with custom location without trailing '/'
    * [x] restart test chainweb-node with custom location without trailing '/'

### CODE REVIEW

File: tools/ea/Ea.hs [hunk 1]
```diff
@@ -163,15 +164,16 @@ genPayloadModule' v tag cwTxs =
     withBlockHeaderDb rocks v cid $ \bhdb -> do
         let logger = genericLogger Warn TIO.putStrLn
         pdb <- newPayloadDb
-        T2 payloadWO _ <- withSqliteDb v cid logger Nothing Nothing False $ \env ->
-            initPactService' v cid logger bhdb pdb env defaultPactServiceConfig $
-                execNewGenesisBlock noMiner (V.fromList cwTxs)
+        withSystemTempDirectory "ea-pact-db" $ \pactDbDir -> do
```
Comment (slpopejoy) [inline tools/ea/Ea.hs:167]: Did you run Ea?

Comment (larskuhtz) [reply tools/ea/Ea.hs:167]: done

**Comment by larskuhtz** [comment]
This code will be removed in chainweb-2.2 (or at least chainweb-2.3).

I tested it manually for several different scenarios. But it would be good if someone else could test it, too.

Not sure if we need a unit tests -- it would be quite a bit of work to write it, given all the file system interactions and corner cases.

File: minimal-config.yaml [hunk 2]
```diff
@@ -90,4 +90,4 @@ logging:
       - key: component
         value: local-handler
         level: info
-    default: error
+    default: error
```
Comment (slpopejoy) [inline minimal-config.yaml:93]: ?

File: node/ChainwebNode.hs [hunk 3]
```diff
@@ -492,3 +517,149 @@ main = do
                 node conf logger
   where
     timeFormat = iso8601DateFormat (Just "%H:%M:%SZ")
+
+-- -------------------------------------------------------------------------- --
+-- Database Director Migration
```
Comment (slpopejoy) [inline node/ChainwebNode.hs:522]:
```suggestion
-- Database Directory Migration
```

File: minimal-config.yaml (hunk 2 above)
Comment (emilypi) [reply minimal-config.yaml:93]: Probably an EOF?
